modules/opengl/glsl/utils/isosurface.glsl

```diff
@@ -91,18 +97,31 @@ vec4 drawIsosurface(in vec4 curResult, in float isovalue, in vec4 isosurfaceColo
         }
 
         vec3 isoposWorld = (volumeParameters.textureToWorld * vec4(isopos, 1.0)).xyz;
-        color.rgb = APPLY_LIGHTING(lighting, color.rgb, color.rgb, vec3(1.0),
+        isocolor.rgb = APPLY_LIGHTING(lighting, isocolor.rgb, isocolor.rgb, vec3(1.0),
                            isoposWorld, -gradient, toCameraDir);
 #endif // SHADING_ENABLED
 
-        // blend surface color and adjust first-hit depth if necessary
+        // apply compositing of volumetric media from last sampling position up till isosurface
+        vec4 isoSample = mix(voxel, previousVoxel, a);
```

Comment:
Use isoValue directly?
mix(voxel, previousVoxel, a) must be equal to isoValue

Reply:
yes, that's correct.